### DISCUSSION TITLE
Support for Bestway 58259 pool heater

### DISCUSSION THREAD

**Discussion by noerstad** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#745] [Q&A]
I am looking to smartify my Bestway 58259 pool heater
Does anyone know if this is supported? 
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6042168/463a8274-7ea3-4203-9294-75c1933de8ec)

58259 only has one button to turn on heater, but also has a surge protection (?) on the cable. This means that I can not use a simple smart relay on the power socket itself to turn on and off. 

Let me know if anyone needs more information or pictures

**Comment by visualapproach** [comment]
Does it come apart without destroying it? In that case you could post pictures of its guts. Those pumps are not supported but it would be fun to see if it's the same design.

**Reply by noerstad** [reply]
![IMG_3150](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6042168/3bdfa92a-0cfa-4ed9-bea6-f58f747447a3)
![IMG_3149](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6042168/25ec9b34-fd58-4959-aa3b-0dc936a6be30)
![IMG_3151](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/6042168/aa2f7d7f-a6ca-41ac-85c3-17d77574e843)

**Reply by noerstad** [reply]
I’m getting the hardware, and will then see if there are any adjustment that needs to be done

**Reply by visualapproach** [reply]
Thanks for the pictures. Might work.

**Reply by noerstad** [reply]
Ok, so got the hardware pre-made. Mounted it between the pump and display using the connectors.

Does not seem to get power. The display does not turn on, and it does not seem that the pcb is booting either (no power?).

Any ideas on how to debug further?

**Reply by visualapproach** [reply]
Measure which wires is 5V and GND and make sure they are correct.

**Reply by noerstad** [reply]
Strugglig to measure properly. Not sure why. Will seek help on that. 

In the meanwhile, I see the original cable has markings. Does this say anything?
Attached is picture of cable and connection to board, as well as correlation with my module

![IMG_3229](https://github.com/user-attachments/assets/ee855366-9110-4595-8d68-ef2e36a5d9cd)
![IMG_3228](https://github.com/user-attachments/assets/b2127966-57db-43b3-9f5b-b4477bead63b)
![IMG_3227](https://github.com/user-attachments/assets/de376243-783f-4eea-9108-36aef1338838)

**Reply by visualapproach** [reply]
The markings are arbitrary and doesn't mean anything. Other than a way to keep track of which wire is which when looking at both ends of the cable.

**Comment by heggico** [comment]
So, I'm trying to figure out if this is possible.
Few issues I found so far:
+5v and Gnd are reversed on the connector. From the side it seems to be GND and then 5V

Then checking with an osciloscope:
Pin 3 (counting the GND pin as 1): this has "activity" when the buzzer is active. So seems to be audio
Then I have 2 with constant data
Then one pin seems to be constant high (5V)

So I connected only my d1-mini to it, reversing polarity to power it.

When comparing input and output signals, the don't match at all. When connecting the display doesn't light up.

I haven't really reverse engineered anything like this before, so i'll keep working on it for a bit. But do you have any tips? Can I debug what I receive an each pin?

Hopefully you have some tips, either way, I'll keep working on it as it seems like a fun project to figure out.

**Reply by visualapproach** [reply]
Cool, there are several ways to reverse engineer stuff. One way is tolook at the ICs involved and search for datasheets. That's not the way I did it on this project but I got things confirmed when other people did it.
You have already made some progress by checking what's happening on all the wires and it appears we have two data lines. Usually things use one of three protocols:
SPI, I2C or UART.
SPI is normally 4 wires- chip select, clock, mosi and miso. But there is variants with bi-directional data lines which is the case on the 6-wire pumps.
I2C have a data and a clock line.
UART have Tx and a Rx line.

Clock lines is driving all the data lines and is necessarily the most active line. Like a square wave, possibly with pauses when no data is sent.
UART don't have a clock but relies on a predefined timing.
If your pump is UART you should only see one of the data lines having activity when disconnecting the display. Thats the Tx on the "CIO".
/visualapproach

**Reply by heggico** [reply]
Yeah I do have some experience with uart/I2C. My scope has a software decoder for both, but I had to get that installed first.

I'll see if I can get it to work! I have used this project for a few years now in a lay-z spa and its working great. Seeing If I can modify it to this sperate heater could be fun.

Thanks for the tips!

**Reply by visualapproach** [reply]
Yeah cracking things is fun. Just post if you wanna discuss details

**Reply by noerstad** [reply]
Nice, @heggico ! The power part is as far as I got before I parked the project (and forgot about it until now) :D
Would love to see what you find out
